tests/semantic_operator_verification_test.py

```
@@ -0,0 +1,230 @@
+            ),
+        ]
+
+    @pytest.mark.asyncio
+    async def test_semantic_operator_usage_tracking(self, mock_llm_provider, sample_population):
+        """Test that semantic operator usage is tracked in metrics."""
+        # Create GA with semantic operators
+        ga = GeneticAlgorithm(llm_provider=mock_llm_provider)
+        
+        # Mock the fitness evaluator
+        with patch.object(ga.fitness_evaluator, 'evaluate_population', new_callable=AsyncMock) as mock_eval:
+            with patch.object(ga.fitness_evaluator, 'calculate_population_diversity', new_callable=AsyncMock) as mock_diversity:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)